frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,469 @@
+/* eslint-disable no-undef */
+
+import * as _ from 'lodash-es';
+import * as classNames from 'classnames';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { coFetchJSON } from '../co-fetch';
+import { alertRuleState } from '../module/monitoring';
+import k8sActions from '../module/k8s/k8s-actions';
+import { MonitoringRoutes, connectToURLs } from '../monitoring';
+import store from '../redux';
+import { UIActions } from '../ui/ui-actions';
+import { ColHead, List, ListHeader, ResourceRow, TextFilter } from './factory';
+import { CheckBoxes } from './row-filter';
+import { SafetyFirst } from './safety-first';
+import { BreadCrumbs, history, NavTitle, SectionHeading, StatusBox, Timestamp, withFallback } from './utils';
+import { formatDuration } from './utils/datetime';
+
+const AlertModel = {
+  kind: 'Alert',
+  label: 'Alert',
+  path: 'alerts',
+  abbr: 'AL',
+};
+export const AlertModelReference = `monitoring:v1:${AlertModel.kind}`;
+
+const AlertRuleModel = {
+  kind: 'AlertRule',
+  label: 'Alert Rule',
+  path: 'alertrules',
+  abbr: 'AR',
+};
+export const AlertRuleModelReference = `monitoring:v1:${AlertRuleModel.kind}`;
+
+const reduxID = 'monitoringRules';
+
+const detailsURL = (model, name, labels) => `/k8s/cluster/${model.path}/${name}?${_.map(labels, (v, k) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`).join('&')}`;
+
+const ExternalLink = ({href, text}) => <a className="co-external-link" href={href} target="_blank" rel="noopener noreferrer">{text}</a>;
+
+const ResourceIcon = props => {
+  const {className, model} = props;
+  return <span className={classNames(`co-m-resource-icon co-m-resource-${model.kind.toLowerCase()}`, className)} title={model.label}>{model.abbr}</span>;
+};
+
+const stateToIconClassName = {
+  firing: 'fa fa-bell alert-firing',
+  pending: 'fa fa-exclamation-triangle alert-pending',
+};
+
+const State: React.SFC<StateProps> = ({state}) => {
+  if (state === 'inactive') {
+    return <span className="text-muted">{_.startCase(state)}</span>;
+  }
+  const klass = stateToIconClassName[state];
+  return klass ? <React.Fragment><i className={klass} aria-hidden="true"></i> {_.startCase(state)}</React.Fragment> : null;
+};
+
+const Annotation = ({children, title}) => _.isNil(children)
+  ? null
+  : <React.Fragment><dt>{title}</dt><dd>{children}</dd></React.Fragment>;
+
+const getURLSearchParams = () => {
+  const labels = {};
+  const params: any = new URLSearchParams(window.location.search);
+  for (let [k, v] of params.entries()) {
+    labels[k] = v;
+  }
+  return labels;
+};
+
+class AlertsPageWrapper extends SafetyFirst<AlertsPageWrapperProps, null> {
```

Comment:
These components need unit tests.

Reply:
I added a few unit tests, but I'd like to leave the e2e tests for a follow up PR.